### DISCUSSION TITLE
Bridge Infill / Internal Bridge Infill Proposal

### DISCUSSION THREAD

**Discussion by Durahl** [supermerill/SuperSlicer#1654] [Community contribution]
Greetings!

I was wondering if there'd be a setting akin to the topmost Layer having only one perimeter `Print Settings` > `Perimeters & Shell` > `Quality `> `Only one perimeter on Top surfaces` but for `Bridge Infill` and instead of reducing the Perimeters **TO** 1 reducing it **BY** 1 replacing the area with the Bridge Infill? 🤔

**Stock behavior:**
![Stock](https://i.imgur.com/X9cA7A1.gif)
The amount of perimeters remains 4 which kinda works at normal speeds but is terrible when going fast ( on a Voron 2.4 ) with the ends of the Infill often enough not connecting with the perimeters over the gaps when using low Internal Infill values.

**Proposal behavior:**
![Proposal](https://i.imgur.com/gAXuYVI.gif)
I'd be hoping for a mechanic that would automatically subtract one perimeter for as long as there's at least two ( in the case of 6 Perimeters it would be doing 5 or with 4 it'd be doing 3 ) and have the Bridge Infill move into the resulting gap effectively sandwiching it between the previous and the following layer. ( For visibility demonstration I reduced the amount of perimeters in the animation from 4 to 2 instead of the proposed 3 )

**DIY Attempt ( doesn't work ):**
![Not Working](https://i.imgur.com/sADFhR1.png)

**Comment by supermerill** [comment]
I already think of it 
https://github.com/supermerill/SuperSlicer/issues/140
My problem lies in identifying these areas and rerouting the last perimeter. geometries can be complex:
![image](https://user-images.githubusercontent.com/6536403/137877476-ab4b403b-3604-409a-a691-2681832f4d89.png)

**Comment by Durahl** [comment]
No doubt about that, I for one wouldn't even know where to start 😃 I just thought of using whatever is being used for the generating the One Perimeter Top Layer feature could also be used for that Proposal - I guess I'm too simple minded 😁

**Reply by supermerill** [reply]
A part of the problem lies with the fact that perimeters are drawn first and then we identify infill regions inside the remnant areas.
To get the solid infill areas before printing perimeters means that a new step of identifying solid areas before perimeters needs to be created.

**Comment by neophyl** [comment]
Would using the extra perimeters on alternating layers help ?  That's one of my favourite functions and I pretty much use it as standard with a 2 or 3 perimeter model (to give 3/4 perimeters).  It seems to do a good job locking the normal infill into the walls.

**Reply by supermerill** [reply]
I think that can help, but only if you're lucky to be on the right layer.

**Comment by Mate76PL** [comment]
Hi. Is it possible for the SS to fill the bridge layer at once in one direction? The entire layer in one direction and one line above the bridge and over the support Now it fills in as in the attachment. CURA fills the entire layer in one direction. 
I think an interesting option would also be to be able to set the angle for this layer, as is now the case with bridges.

![screenshot 107 (19)](https://user-images.githubusercontent.com/101880527/159024234-b02f6ea2-5f29-4272-8f1c-c8f101489e6a.jpg)


Cura:
![screenshot 107 (18)](https://user-images.githubusercontent.com/101880527/159022436-eb0eb130-b242-46e4-a825-3eceaf4be17b.jpg)

**Reply by supermerill** [reply]
in infill tab, angle is 'Angle -> bridging' if you want to force it.
if you want an entire layer of bridges (i don't recommend unless you lower the flow drastically), increase the anchor, the line below.

why it's that way:
 * angle of solid infill is rotated 90° each layer to increase the strength.
 * bridge is filled separately because it choose the angle to minimise the distance to travel, and to try to have an anchor at each side of each line. Also, the flow is set to have good bridges, it's different from solid infill bridge that had to not over-extrude..

**Reply by supermerill** [reply]
> I understand, but I often print boxes with a logo. To make the logo clear, I use bridges on the 2nd layer of the model - as you can see in the photo. It doesn't go wrong, but the bridges aren't continuously printed, it just prints a piece, the head jumps to another spot, then goes back to the unfinished bridge. There are visible gaps. I make up for it, of course, with the width of the extrusion and flow but it would be much better if the bridge layer was printed as in Cura "at once".

cura doesn't print bridges, as far as I understand your picture.

did you try my solutions?

**Reply by supermerill** [reply]
> Cura prints the layer without dividing it into a bridge and a support

support? there is no support.

btw, you know, you have a field just below, here \\/ marked "write a reply", you don't need to start a new thread with "write a comment""

**Comment by Mate76PL** [comment]
cura doesn't print bridges, as far as I understand your picture.  - Cura prints the layer without dividing it into a bridge and a support

did you try my solutions? - Yes, looks better. I haven't checked this option, Thanks for your help.
![screenshot 107 (20)](https://user-images.githubusercontent.com/101880527/159088693-8b2b8ad9-41e6-4697-876c-07c8423ed986.jpg)
